### PR TITLE
Refactor: simplify parsing of keyword sequences as Vars

### PR DESCRIPTION
**Pull request by georgesittas**

### CODE REVIEW

File: sqlglot/parser.py [hunk 1]
```diff
@@ -5692,11 +5677,29 @@ def _parse_set(self, unset: bool = False, tag: bool = False) -> exp.Set | exp.Co
...
+        for keywords in continuations or []:
+            if isinstance(keywords, str):
+                keywords = (keywords,)
+
+            if self._match_text_seq(*keywords):
+                option = f"{option} {' '.join(keywords)}"
+                break
+        else:
+            if continuations:
+                self.raise_error(f"Unknown option {option}")
```
Comment (tobymao) [inline sqlglot/parser.py:5700]: is this more strict than it needs to be?

Comment (tobymao) [reply sqlglot/parser.py:5700]: yea

Comment (georgesittas) [reply sqlglot/parser.py:5700]: Hmm maybe it's actually better to keep it, at least for now. I tried the retreat approach but e.g. for T-SQL we try to match a `)` after the option parser is called and that fails so the error message is cryptic - it says we're expecting a `)` but in reality we just failed to parse the option and retreated without consuming the tokens.

Comment (tobymao) [reply sqlglot/parser.py:5700]: k

**Pull request merged**